packages/netlify-cms-backend-azure/src/API.ts

```diff
@@ -0,0 +1,782 @@
+      this.readFile(path, null, { branch }) as Promise<string>,
+      this.isFileExists(path, this.branch),
+    ]);
+
+    return {
+      slug,
+      metaData: { branch, collection, objects: { entry: { path, mediaFiles } }, status },
+      fileData,
+      isModification,
+    };
+  }
+
+  isUnpublishedEntryModification(path: string, branch: string) {
+    return this.readFile(path, null, { branch })
+      .then(() => true)
+      .catch((err: Error) => {
+        if (err.message && err.message === 'Not Found') {
+          return false;
+        }
+        throw err;
+      });
+  }
+
+  /**
+   * Retrieve statuses for a given SHA. Unrelated to the editorial workflow
+   * concept of entry "status". Useful for things like deploy preview links.
+   */
+  async getStatuses(sha: string) {
+    const resp = await this.requestJSON(`${this.endpointUrl}/commits/${sha}/status`);
+    return resp.statuses;
+  }
+
+  async getCommitItems(files: (Entry | AssetProxy)[], branch: string) {
+    const items = await Promise.all(
+      files.map(async file => {
+        const [base64Content, fileExists] = await Promise.all([
+          result(file, 'toBase64', partial(this.toBase64, (file as Entry).raw)),
+          this.isFileExists(file.path, branch),
+        ]);
+        return {
+          action: fileExists ? AzureCommitChangeType.EDIT : AzureCommitChangeType.ADD,
+          base64Content,
+          path: '/' + trim(file.path, '/'),
+        };
+      }),
+    );
+    return items as any[];
+  }
+
+  /**
+   * Store a resource in the target repository.
+   * @param entry
+   * @param mediaFiles
+   * @param options
+   */
+  async persistFiles(entry: Entry | null, mediaFiles: AssetProxy[], options: PersistOptions) {
+    const files = entry ? [entry, ...mediaFiles] : mediaFiles;
+    if (options.useWorkflow) {
+      return this.editorialWorkflowGit(files, entry as Entry, options);
+    } else {
+      const items = await this.getCommitItems(files, this.branch);
+      return this.uploadAndCommit(items, options.commitMessage);
+    }
+  }
+
+  deleteFile(path: string, comment: string, branch = this.branch) {
+    return this.getRef(branch).then((ref: AzureRef) => {
+      const commit = new AzureCommit(comment);
+      commit.changes.delete(path);
+
+      const push = new AzurePush(ref);
+      push.commits.push(commit);
+
+      return this.requestJSON({
+        url: `${this.endpointUrl}/pushes`,
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json; charset=utf-8' },
+        body: JSON.stringify(push),
+      });
+    });
+  }
+
+  generateContentKey(collectionName: string, slug: string) {
+    return generateContentKey(collectionName, slug);
+  }
+
+  contentKeyFromBranch(branch: string) {
+    return branch.substring(`${CMS_BRANCH_PREFIX}/`.length);
+  }
+
+  branchFromContentKey(contentKey: string) {
+    return `${CMS_BRANCH_PREFIX}/${contentKey}`;
+  }
+
+  async getMergeRequests(sourceBranch?: string) {
```

Comment:
We should use "pull requests" naming along this file if that's what the API uses.